sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py

```diff
@@ -184,6 +184,30 @@ def on_request(self, request):
         request.http_request.headers[self._name] = self._credential.key
 
 
+class AzureHttpKeyCredentialPolicy(SansIOHTTPPolicy):
```

Comment:
Instead of introducing a new policy, can we just extend AzureKeyCredentialPolicy to add the scheme setting?

It looks to me this is like AzureKeyCredentialPolicy(credential: "AzureKeyCredential", name="Authorization", auth_scheme: str = "Basic", **kwargs)

Reply:
The header value is built by the concatenation of the scheme and the key. This would require to add logic in the AzureKeyCredential to detect the presence of the scheme parameter and then concatenate it at the beginning with a space. It feels we're adding complexity doing that IMHO.

Reply:
This is what I meant :):

```py
class AzureKeyCredentialPolicy(SansIOHTTPPolicy):
    """Adds a key header for the provided credential.

    :param credential: The credential used to authenticate requests.
    :type credential: ~azure.core.credentials.AzureKeyCredential
    :param str name: The name of the key header used for the credential.
    :raises: ValueError or TypeError
    """

    def __init__(
        self,
        credential: "AzureKeyCredential",
        name: str = "Authorization",
        auth_scheme: str = "",
        **kwargs  # pylint: disable=unused-argument
    ) -> None:
        super(AzureKeyCredentialPolicy, self).__init__()
        self._credential = credential
        if not name:
            raise ValueError("name can not be None or empty")
        if not isinstance(name, str):
            raise TypeError("name must be a string.")
        self._name = name
        self._auth_scheme = auth_scheme + " " if auth_scheme else ""

    def on_request(self, request):
        request.http_request.headers[self._name] = f"{self._auth_scheme}{self._credential.key}"
```

Reply:
I'm not convinced it's better, curious to hear other people thoughts on this :)

Reply:
I updated my PR to this, however I don't think we should have a default value for `name`

Reply:
> I updated my PR to this, however I don't think we should have a default value for `name`

The default value of `name` is to be compatible with your original PR. :)

def __init__(
        self, credential: "AzureKeyCredential", auth_scheme: str = "Basic", **kwargs  # pylint: disable=unused-argument
    ) -> None:

if user does not specify `name`, we put it into headers["Authorization"].

Reply:
I think extending AzureKeyCredential like this looks okay - do we know of any other scenarios apart from EventGrid where this would be needed? How are other languages approaching this?

I was wondering whether conceptually `AzureKeyCredential` was intended to represent a particular header format - but that seems unlikely given that we already allow for customizing the header name...

Reply:
@annatisch there is an email thread I'm adding you that shows all languages are the same page to add a new parameter.